tee-worker/omni-executor/native-task-handler/src/lib.rs

```diff
@@ -192,11 +196,28 @@ async fn handle_native_task<
 				return;
 			};
 			let auth_options = AuthOptions { expires_at: current_block + AUTH_TOKEN_EXPIRATION };
-			let claims = AuthTokenClaims::new(
-				sender.hash().to_string(),
-				AUTH_TOKEN_SESSION_TYPE.to_string(),
-				auth_options,
-			);
+			let claims = match sender {
+				Identity::Email(ref identity_string) => {
+					let Ok(email) = std::str::from_utf8(identity_string.inner_ref()) else {
+						let response =
+							NativeTaskResponse::Err(NativeTaskError::InvalidMemberIdentity);
+						if response_sender.send(response.encode()).is_err() {
+							log::error!("Failed to send response");
+						}
+						return;
+					};
+					AuthTokenClaims::new(
+						email.to_string(),
+						AUTH_TOKEN_ID_TYPE.to_string(),
+						auth_options,
+					)
+				},
+				_ => AuthTokenClaims::new(
+					sender.hash().to_string(),
+					AUTH_TOKEN_ID_TYPE.to_string(),
+					auth_options,
+				),
+			};
```

Comment:
Just to make sure that I understand it correctly, this part (`RequestAuthToken`) is actually not used in the pumpx flow, isn't it?

Reply:
Correct